test/demo/contextChannels/join.inttest.ts

```diff
@@ -0,0 +1,446 @@
+
+        // Open our app
+        await (openFunction as (() => Promise<void>))();
+
+        // Check we received a channel-changed event
+        await expect(listener.getReceivedEvents()).resolves.toEqual([{
+            type: 'channel-changed',
+            identity: {uuid: (appIdentity as Identity).uuid, name: (appIdentity as Identity).name},
+            channel: {id: 'default', type: 'default'},
+            previousChannel: null
+        }]);
+    }, appStartupTime);
+
+    test('Event is not received when a non-FDC3 app starts', async () => {
+        // Set up our listener
+        const listener = await fdc3Remote.addEventListener(listeningApp, 'channel-changed');
+
+        // Start our non-FDC3 app
+        await fin.Application.startFromManifest(testAppNotFdc3.manifestUrl);
+
+        // Check no event is received
+        await expect(listener.getReceivedEvents()).resolves.toEqual([]);
+    });
+
+    test('Event is received when an FDC3 app quits', async () => {
+        // Open our FDC3 app ahead of setting up our listener
+        await fdc3Remote.open(testManagerIdentity, testAppInDirectory2.name);
+
+        // Set up our listener then quit the app
+        const listener = await fdc3Remote.addEventListener(listeningApp, 'channel-changed');
+        await fin.Application.wrapSync(testAppInDirectory2).quit(true);
+
+        // Check we received a channel-changed event
+        await expect(listener.getReceivedEvents()).resolves.toEqual([{
+            type: 'channel-changed',
+            identity: {uuid: testAppInDirectory2.uuid, name: testAppInDirectory2.name},
+            channel: null,
+            previousChannel: {id: 'default', type: 'default'}
+        }]);
+    }, appStartupTime);
+});
+
+describe('When attempting to join a channel', () => {
+    let blueChannel: RemoteChannel;
+
+    beforeEach(async () => {
+        blueChannel = await fdc3Remote.getChannelById(testManagerIdentity, 'blue');
+    });
+
+    test('If an invalid identity is provided, a TypeError is thrown', async () => {
+        const invalidIdentity = {irrelevantProperty: 'irrelevantValue'} as unknown as Identity;
+
+        await expect(blueChannel.join(invalidIdentity)).
+            rejects.
+            toThrowError(new TypeError(`${JSON.stringify(invalidIdentity)} is not a valid Identity`));
+    });
+
+    test('If an identity for a window that does not exist is provided, an FDC3 error is thrown', async () => {
+        const nonExistentWindowIdentity: Identity = {uuid: 'does-not-exist', name: 'does-not-exist'};
+
+        await expect(blueChannel.join(nonExistentWindowIdentity)).
+            toThrowFDC3Error(
+                IdentityError.WindowWithIdentityNotFound,
+                `No connection to FDC3 service found from window with identity: ${JSON.stringify(nonExistentWindowIdentity)}`
+            );
+    });
+
+    describe('When a non-FDC3 has been started', () => {
```

Comment:
Some weird wording here in this block and the one below...
- "When a non-FDC3 _app_ has been started"
- "If _a_ non-FDC3 app ..."
- (etc)